Build failure: medusa

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

- build package medusa
- get errors from libssh2

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/319495835

### Relevant log output

```console
error: builder for '/nix/store/mvx213zdjhsp8cx8a7hg6wfv0bzjyxq5-medusa-2.3.drv' failed with exit code 2;
       last 50 log lines:
       > gcc -DHAVE_CONFIG_H -I. -I../..  -I../../src  -g -DDEBUG -fPIC -I/usr/include -I/usr/local/include -I/usr/include/postgresql -I/usr/include/pgsql -I/usr/include/afpfs-ng -I/usr/include/freerdp2 -I/usr/include/winpr2 -I/usr/include/smb2 -D_GNU_SOURCE  -g -O2 -c -o rsh.o rsh.c
       > gcc  -g -O2 -fPIC -rdynamic  -o rsh.mod rsh.o ../medusa-trace.o -lssh2 -shared -ldl -lpthread  -lssl -lcrypto -lssl -lcrypto -ldl -lrt -lm
       > gcc -DHAVE_CONFIG_H -I. -I../..  -I../../src  -g -DDEBUG -fPIC -I/usr/include -I/usr/local/include -I/usr/include/postgresql -I/usr/include/pgsql -I/usr/include/afpfs-ng -I/usr/include/freerdp2 -I/usr/include/winpr2 -I/usr/include/smb2 -D_GNU_SOURCE  -g -O2 -c -o smbnt.o smbnt.c
       > gcc -DHAVE_CONFIG_H -I. -I../..  -I../../src  -g -DDEBUG -fPIC -I/usr/include -I/usr/local/include -I/usr/include/postgresql -I/usr/include/pgsql -I/usr/include/afpfs-ng -I/usr/include/freerdp2 -I/usr/include/winpr2 -I/usr/include/smb2 -D_GNU_SOURCE  -g -O2 -c -o smbnt-smb1.o smbnt-smb1.c
       > gcc -DHAVE_CONFIG_H -I. -I../..  -I../../src  -g -DDEBUG -fPIC -I/usr/include -I/usr/local/include -I/usr/include/postgresql -I/usr/include/pgsql -I/usr/include/afpfs-ng -I/usr/include/freerdp2 -I/usr/include/winpr2 -I/usr/include/smb2 -D_GNU_SOURCE  -g -O2 -c -o smbnt-smb2.o smbnt-smb2.c
       > gcc -DHAVE_CONFIG_H -I. -I../..  -I../../src  -g -DDEBUG -fPIC -I/usr/include -I/usr/local/include -I/usr/include/postgresql -I/usr/include/pgsql -I/usr/include/afpfs-ng -I/usr/include/freerdp2 -I/usr/include/winpr2 -I/usr/include/smb2 -D_GNU_SOURCE  -g -O2 -c -o hmacmd5.o hmacmd5.c
       > gcc  -g -O2 -fPIC -rdynamic  -o smbnt.mod smbnt.o smbnt-smb1.o smbnt-smb2.o hmacmd5.o ../medusa-trace.o -lssh2 -shared -ldl -lpthread  -lssl -lcrypto -lssl -lcrypto -ldl -lrt -lm
       > gcc -DHAVE_CONFIG_H -I. -I../..  -I../../src  -g -DDEBUG -fPIC -I/usr/include -I/usr/local/include -I/usr/include/postgresql -I/usr/include/pgsql -I/usr/include/afpfs-ng -I/usr/include/freerdp2 -I/usr/include/winpr2 -I/usr/include/smb2 -D_GNU_SOURCE  -g -O2 -c -o smtp.o smtp.c
       > gcc  -g -O2 -fPIC -rdynamic  -o smtp.mod smtp.o ntlm.o ../medusa-trace.o -lssh2 -shared -ldl -lpthread  -lssl -lcrypto -lssl -lcrypto -ldl -lrt -lm
       > gcc -DHAVE_CONFIG_H -I. -I../..  -I../../src  -g -DDEBUG -fPIC -I/usr/include -I/usr/local/include -I/usr/include/postgresql -I/usr/include/pgsql -I/usr/include/afpfs-ng -I/usr/include/freerdp2 -I/usr/include/winpr2 -I/usr/include/smb2 -D_GNU_SOURCE  -g -O2 -c -o smtp-vrfy.o smtp-vrfy.c
       > gcc  -g -O2 -fPIC -rdynamic  -o smtp-vrfy.mod smtp-vrfy.o ../medusa-trace.o -lssh2 -shared -ldl -lpthread  -lssl -lcrypto -lssl -lcrypto -ldl -lrt -lm
       > gcc -DHAVE_CONFIG_H -I. -I../..  -I../../src  -g -DDEBUG -fPIC -I/usr/include -I/usr/local/include -I/usr/include/postgresql -I/usr/include/pgsql -I/usr/include/afpfs-ng -I/usr/include/freerdp2 -I/usr/include/winpr2 -I/usr/include/smb2 -D_GNU_SOURCE  -g -O2 -c -o snmp.o snmp.c
       > gcc  -g -O2 -fPIC -rdynamic  -o snmp.mod snmp.o ../medusa-trace.o -lssh2 -shared -ldl -lpthread  -lssl -lcrypto -lssl -lcrypto -ldl -lrt -lm
       > gcc -DHAVE_CONFIG_H -I. -I../..  -I../../src  -g -DDEBUG -fPIC -I/usr/include -I/usr/local/include -I/usr/include/postgresql -I/usr/include/pgsql -I/usr/include/afpfs-ng -I/usr/include/freerdp2 -I/usr/include/winpr2 -I/usr/include/smb2 -D_GNU_SOURCE  -g -O2 -c -o ssh.o ssh.c
       > ssh.c: In function 'initModule':
       > ssh.c:261:9: warning: 'libssh2_banner_set' is deprecated: since libssh2 1.4.0. Use libssh2_session_banner_set() [8;;https://gcc.gnu.org/onlinedocs/gcc-15.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations8;;]
       >   261 |         if ( libssh2_banner_set(session, _psSessionData->szBannerMsg) ) {
       >       |         ^~
       > In file included from ssh.c:46:
       > /nix/store/dp3sg21kr75b2059s94g3a2vg8dfajmv-libssh2-1.11.1-dev/include/libssh2.h:666:17: note: declared here
       >   666 | LIBSSH2_API int libssh2_banner_set(LIBSSH2_SESSION *session,
       >       |                 ^~~~~~~~~~~~~~~~~~
       > ssh.c:287:11: warning: 'libssh2_session_startup' is deprecated: since libssh2 1.2.8. Use libssh2_session_handshake() [8;;https://gcc.gnu.org/onlinedocs/gcc-15.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations8;;]
       >   287 |           if (libssh2_session_startup(session, hSocket)) {
       >       |           ^~
       > /nix/store/dp3sg21kr75b2059s94g3a2vg8dfajmv-libssh2-1.11.1-dev/include/libssh2.h:670:17: note: declared here
       >   670 | LIBSSH2_API int libssh2_session_startup(LIBSSH2_SESSION *session, int sock);
       >       |                 ^~~~~~~~~~~~~~~~~~~~~~~
       > ssh.c: In function 'tryLogin':
       > ssh.c:442:21: error: assignment to 'void (*)(void)' from incompatible pointer type 'void (*)(const char *, int,  const char *, int,  int,  const LIBSSH2_USERAUTH_KBDINT_PROMPT *, LIBSSH2_USERAUTH_KBDINT_RESPONSE *, void **)' [8;;https://gcc.gnu.org/onlinedocs/gcc-15.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
       >   442 |   pResponseCallback = response_callback;
       >       |                     ^
       > ssh.c:398:6: note: 'response_callback' declared here
       >   398 | void response_callback(const char* name, int name_len, const char* instruction, int instruction_len, int num_prompts, const LIBSSH2_USERAUTH_KBDINT_PROMPT* prompts, LIBSSH2_USERAUTH_KBDINT_RESPONSE* responses, void **abstract)
       >       |      ^~~~~~~~~~~~~~~~~
       > ssh.c:495:11: error: passing argument 4 of 'libssh2_userauth_keyboard_interactive_ex' from incompatible pointer type [8;;https://gcc.gnu.org/onlinedocs/gcc-15.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
       >   495 |       if (libssh2_userauth_keyboard_interactive(session, szLogin, pResponseCallback) )
       >       |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       >       |           |
       >       |           void (*)(void)
       > /nix/store/dp3sg21kr75b2059s94g3a2vg8dfajmv-libssh2-1.11.1-dev/include/libssh2.h:798:42: note: expected 'void (*)(const char *, int,  const char *, int,  int,  const LIBSSH2_USERAUTH_KBDINT_PROMPT *, LIBSSH2_USERAUTH_KBDINT_RESPONSE *, void **)' but argument is of type 'void (*)(void)'
       >   798 |                                          LIBSSH2_USERAUTH_KBDINT_RESPONSE_FUNC
       >       |                                          ^
       > make[3]: *** [Makefile:781: ssh.o] Error 1
       > make[3]: Leaving directory '/build/source/src/modsrc'
       > make[2]: *** [Makefile:418: all-recursive] Error 1
       > make[2]: Leaving directory '/build/source/src'
       > make[1]: *** [Makefile:446: all-recursive] Error 1
       > make[1]: Leaving directory '/build/source'
       > make: *** [Makefile:344: all] Error 2
       For full logs, run 'nix log /nix/store/mvx213zdjhsp8cx8a7hg6wfv0bzjyxq5-medusa-2.3.drv'.
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.3-xanmod1, Red Flake, rolling, rolling`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/pascal/.config/nix/nix.conf:/home/pascal/.config/kdedefaults/nix/nix.conf:/nix/store/2z1c2l23xph0nsi6zrjgijllgd60b78v-plasma-workspace-6.5.4/etc/xdg/nix/nix.conf:/nix/store/bii6jwr2jb3qs8vzn7n5qnwi1xnckbmj-kglobalacceld-6.5.4/etc/xdg/nix/nix.conf:/nix/store/nfrr23vi3iyphga4nnqdbnccdif51g0c-baloo-6.21.0/etc/xdg/nix/nix.conf:/home/pascal/.config/kdedefaults/nix/nix.conf:/etc/xdg/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/pascal/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/pascal/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/pascal/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/pascal/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/m7fmmz7p9c4v9vzr90blpncqbc1nc4zr-lix-2.93.3/share`
 - nixpkgs: `/nix/store/lv4kcr2y4b9k4ynhkga157507zxf372b-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/me/medusa/package.nix
+++ b/pkgs/by-name/me/medusa/package.nix
@@ -16,7 +16,7 @@ stdenv.mkDerivation (finalAttrs: {
     owner = "jmk-foofus";
     repo = "medusa";
     tag = finalAttrs.version;
-    hash = "sha256-devirQMmS8mtxT5H5XafRRvCyfcvwoWxtTp0V1SJeSM=";
+    hash = "sha256-ftn5RBE3NYfjXLq8Gm92sbFW+M925BDuL/VmwfPYXpo=";
   };
 
   outputs = [

